lib/overcommit/hook/pre_push/git_lfs.rb

```diff
@@ -0,0 +1,21 @@
+module Overcommit::Hook::PrePush
+  # Invokes Git LFS command that uploads files tracked by Git LFS to the LFS storage
+  #
+  # @see https://git-lfs.github.com/
+  class GitLfs < Base
+    def run
+      result = execute(['command', '-v', 'git-lfs'])
+      unless result.success?
+        return :warn, 'This repository is configured for Git LFS but \'git-lfs\' ' \
+        'was not found on your path.\nIf you no longer wish to use Git LFS, ' \
+        'disable this hook by removing or setting \'enabled: false\' for GitLFS ' \
+        'hook in your .overcommit.yml file'
+      end
+
+      result = execute(['git', 'lfs', 'pre-push', remote_name, remote_url])
+      return :fail, result.stderr unless result.success?
+
+      :pass
```

Comment:
> Would it make sense to instead of actually uploading just check if there is something to upload?

Not really, you'd just want it to be uploaded just with `git push`, without additionally calling `git lfs ...`. Also, I'd not make overcommit behaviour different from standard (meaning what is installed by LFS client by default).

> It's well outside the scope of what you're trying to accomplish with this change.

I totally agree, I was wondering if there's a way to do that, but if not, it's fine. It might be just mentioned in the doc since GitLfs hook is disabled by default and requires the user to add it in the config yaml.